zerver/actions/message_edit.py

```diff
@@ -765,50 +766,55 @@ def user_info(um: UserMessage) -> Dict[str, Any]:
         assert stream_being_edited is not None
         assert topic_name is not None or new_stream is not None
 
-        for muting_user in get_users_muting_topic(stream_being_edited.id, orig_topic_name):
+        for user_topic in get_users_with_user_topic_visibility_policy(
+            stream_being_edited.id, orig_topic_name
+        ):
             # TODO: Ideally, this would be a bulk update operation,
             # because we are doing database operations in a loop here.
             #
             # This loop is only acceptable in production because it is
-            # rare for more than a few users to have muted an
-            # individual topic that is being moved; as of this
+            # rare for more than a few users to have visibility_policy
+            # set for an individual topic that is being moved; as of this
             # writing, no individual topic in Zulip Cloud had been
             # muted by more than 100 users.
 
-            if new_stream is not None and muting_user.id in delete_event_notify_user_ids:
+            if (
+                new_stream is not None
+                and user_topic.user_profile_id in delete_event_notify_user_ids
+            ):
                 # If the messages are being moved to a stream the user
                 # cannot access, then we treat this as the
                 # messages/topic being deleted for this user. This is
                 # important for security reasons; we don't want to
                 # give users a UserTopic row in a stream they cannot
-                # access.  Unmute the topic for such users.
+                # access. Remove the user topic rows for such users.
                 do_set_user_topic_visibility_policy(
-                    muting_user,
+                    user_topic.user_profile,
                     stream_being_edited,
                     orig_topic_name,
                     visibility_policy=UserTopic.VisibilityPolicy.INHERIT,
                 )
             else:
-                # Otherwise, we move the muted topic record for the
-                # user, but removing the old topic mute and then
-                # creating a new one.
+                # Otherwise, we move the user topic record for the
+                # user, but removing the old topic visibility_policy
+                # and then creating a new one.
                 do_set_user_topic_visibility_policy(
```

Comment:
It might be more readable to do `new_visibility_policy = user_topic.visibility_policy` before doing any mutation in this next block, since then one doesn't have to think about the possibility of variable reuse issues. (We don't pass the `user_topic` object to `do_set_user_topic_visibility_policy`, but one can imagine a refactor where we do and thus the original value ends up getting mutated).

Reply:
Updated with `new_visibility_policy = user_topic.visibility_policy`